packages/front-end/components/Experiment/CustomFieldDisplay.tsx

```diff
@@ -0,0 +1,128 @@
+import React, { FC, useState } from "react";
+import {
+  CustomExperimentField,
+  ExperimentInterfaceStringDates,
+} from "back-end/types/experiment";
+import { useForm } from "react-hook-form";
+import { CustomField } from "back-end/types/organization";
+import { useUser } from "@/services/UserContext";
+import { useCustomFields } from "../../services/experiments";
+import HeaderWithEdit from "../Layout/HeaderWithEdit";
+import Modal from "../Modal";
+import { useAuth } from "../../services/auth";
+import CustomFieldInput from "./CustomFieldInput";
+
+const CustomFieldDisplay: FC<{
+  label?: string;
+  canEdit?: boolean;
+  mutate?: () => void;
+  experiment: ExperimentInterfaceStringDates;
+}> = ({ label = "Additional Fields", experiment, canEdit = true, mutate }) => {
+  const [editModal, setEditModal] = useState(false);
+  const customFields = useCustomFields();
+  const customFieldsMap = new Map();
+  const defaultFields: CustomExperimentField = {};
+  if (customFields && customFields.length) {
+    customFields.map((v) => {
+      defaultFields[v.id] =
+        v.type === "boolean"
+          ? JSON.stringify("")
+          : v.type === "multiselect"
+          ? JSON.stringify([])
+          : "";
+      customFieldsMap.set(v.id, v);
+    });
+  }
+
+  const { license } = useUser();
+  const form = useForm<Partial<ExperimentInterfaceStringDates>>({
+    defaultValues: {
+      customFields: experiment?.customFields || defaultFields,
+    },
+  });
+  const { apiCall } = useAuth();
+
+  if (customFields && customFields.length) {
+    return (
+      <>
+        {editModal && (
+          <Modal
+            header={"Edit Custom Fields"}
+            open={editModal}
+            close={() => {
+              setEditModal(false);
+            }}
+            size="lg"
+            submit={form.handleSubmit(async (value) => {
+              const data = { ...value };
+
+              await apiCall(`/experiment/${experiment.id}`, {
+                method: "POST",
+                body: JSON.stringify(data),
+              });
+
+              if (mutate) mutate();
+            })}
+            cta="Save"
+          >
+            {license ? (
```

Comment:
Should probably use the `hasCommercialFeature("your-feature-name")` check, as well as the `<PremiumTooltip commercialFeature="your-feature-name">` wrapper on the label text